harness/unittests/test_judge_anchored.py

```
@@ -0,0 +1,225 @@
+        self.assertEqual(_evaluate_arithmetic_expression("(((2 + 3) * 4) - 1)"), 19.0)
+
+    def test_zero_operations(self) -> None:
+        """Test operations involving zero."""
+        self.assertEqual(_evaluate_arithmetic_expression("0 + 0"), 0.0)
+        self.assertEqual(_evaluate_arithmetic_expression("0 * 5"), 0.0)
+        self.assertEqual(_evaluate_arithmetic_expression("5 * 0"), 0.0)
+        self.assertEqual(_evaluate_arithmetic_expression("0 - 0"), 0.0)
+
+    def test_fractional_results(self) -> None:
+        """Test expressions that produce fractional results."""
+        self.assertEqual(_evaluate_arithmetic_expression("1 / 3"), 1.0 / 3.0)
```

Ruff (0.14.3)

Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)